test/unit/job/test_user_metrics.py

```diff
@@ -173,101 +192,78 @@ def test_metric_spec_load_invalid(metric_specs: Any):
         MetricSpec.load(metric_specs)
 
 
-def test_read_metrics_specs(disable_metric_repository):
+def test_read_metrics_specs_as_conf_exception(disable_metric_repository, mocker):
     # Mocking the AutosubmitConfig and Job objects
-    as_conf = MagicMock()
-    job = MagicMock()
-
-    as_conf.get_section.return_value = [
-        {"NAME": "metric1", "FILENAME": "file1"},
-        {
-            "NAME": "invalid metric",
-        },
-        {
-            "NAME": "metric2",
-            "FILENAME": "file2",
-            "SELECTOR": {"TYPE": "JSON", "KEY": "key1.key2.key3"},
-        },
-    ]
-    as_conf.deep_normalize = lambda x: x
+    as_conf = mocker.MagicMock()
+    as_conf.get_section.side_effect = ValueError
+    job = mocker.MagicMock()
 
     # Do the read test
     user_metric_processor = UserMetricProcessor(as_conf, job)
-    metric_specs = user_metric_processor.read_metrics_specs()
-
-    assert len(metric_specs) == 2
 
-    assert metric_specs[0].name == "metric1"
-    assert metric_specs[0].filename == "file1"
-    assert metric_specs[0].selector.type == MetricSpecSelectorType.TEXT
-    assert metric_specs[0].selector.key is None
+    with pytest.raises(ValueError) as cm:
+        user_metric_processor.read_metrics_specs()
 
-    assert metric_specs[1].name == "metric2"
-    assert metric_specs[1].filename == "file2"
-    assert metric_specs[1].selector.type == MetricSpecSelectorType.JSON
-    assert metric_specs[1].selector.key == ["key1", "key2", "key3"]
+    assert 'Invalid or missing metrics section' in str(cm.value)
 
 
-def test_process_metrics(disable_metric_repository):
+def test_process_metrics(disable_metric_repository, mocker):
```

Comment:
```suggestion
def test_process_metrics(disable_metric_repository, mocker: MockerFixture):
```

I might add this to have the type hints

Reply:
Applied, thanks!

Reply:
Huh,

```
ERROR test/unit/job/test_user_metrics.py - NameError: name 'MockerFixture' is not defined
```

I think I will have to import that first. Let me fix that and squash the commits...